h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -207,12 +117,174 @@ class SearchBarController extends Controller {
       }
     };
 
+    let explanationList = [
+      {
+        matchOn: 'user',
+        title: 'user:',
+        explanation: 'search by username',
+      },
+      {
+        matchOn: 'tag',
+        title: 'tag:',
+        explanation: 'search for annotations with a tag',
+      },
+      {
+        matchOn: 'url',
+        title: 'url:',
+        explanation: 'see all annotations on a page',
+      },
+      {
+        matchOn: 'group',
+        title: 'group:',
+        explanation: 'show annotations created in a group you are a member of',
+      },
+    ].map((item)=>{ return Object.assign(item, { type: FACET_TYPE}); });
+
+    // tagSuggestions are made available by the scoped template data.
+    // see search.html.jinja2 for definition
+    let tagSuggestions = window.h && window.h.tagSuggestions;
+
+    let tagsList = (tagSuggestions || []).map((item)=>{
+      return Object.assign(item, {
+        type: TAG_TYPE,
+        title: escapeHtml(item.tag), // make safe
```

Comment:
This is fine, although I'm concerned that there will be a mixup with escaping in future, where something escaped for HTML is incorrectly used in a different context or some transformation gets applied to the string which breaks the escaping (on the Python side, this hazard is avoided by wrapping escaped strings using a special `Markup` class).